src/components/Select/Select.stories.tsx

```diff
@@ -248,6 +346,62 @@ export const UsingChildrenProp: StoryObj = {
 };
 
 export const UsingFunctionChildrenProp: StoryObj = {
+  parameters: {
+    docs: {
+      source: {
+        code: `
+function InteractiveExampleUsingFunctionChildren() {
+  const [selectedOption, setSelectedOption] =
+    React.useState<(typeof exampleOptions)[0]>();
+
+  return (
+    <div className="mb-10 p-8">
+      <Select
+        aria-label="Favorite Animal"
+        as="div"
+        className="w-60"
+        data-testid="dropdown"
+        onChange={setSelectedOption}
+        value={selectedOption}
+      >
+        {({ open }) => (
+          <>
+            <Select.Button
+              // Because we're using a render prop to completely control the styling and icon of the
+              // button, we need to configure this component to render as a Fragment. Otherwise we'd
+              // render two, nested buttons.
+              as={React.Fragment}
+            >
+              {() => (
+                <button
+                  aria-expanded={open}
+                  className={styles['function-children__button']}
+                >
+                  {selectedOption?.label || 'Select'}
+                  <Icon
+                    className="ml-4"
+                    name="filter-list"
+                    purpose="decorative"
+                  />
+                </button>
+              )}
+            </Select.Button>
+            <Select.Options>
+              {exampleOptions.map((option) => (
```

Comment:
in https://chanzuckerberg.github.io/edu-design-system/?path=/story/documentation-guidelines-code-guidelines--page we could advise people use this storybook config when a component has rend props, for example